src/pybind/mgr/dashboard/services/orchestrator.py

```diff
@@ -39,7 +39,8 @@ def list_service_info(self, service_type):
         return completion.result
 
     def available(self):
-        if not Settings.ORCHESTRATOR_BACKEND:
+        _backend = mgr.remote("orchestrator_cli", "_select_orchestrator")
```

Comment:
This fails if the ``orchestrator_cli`` module is disabled.
```
2019-02-08 14:20:53.276 7fc58a110700 -1 no module 'orchestrator_cli'
2019-02-08 14:20:53.284 7fc58a110700  0 mgr[dashboard] [::ffff:127.0.0.1:49738] [GET] [500] [8.668s] [admin] [1.4K] /api/mgr/module/telemetry
2019-02-08 14:20:53.284 7fc58a110700  0 mgr[dashboard] [b'{"status": "500 Internal Server Error", "detail": "The server encountered an unexpected condition which prevented it from fulfilling the request.", "traceback": "Traceback (most recent call last):\\n  File \\"/usr/lib/python3.7/site-packages/cherrypy/_cprequest.py\\", line 670, in respond\\n    response.body = self.handler()\\n  File \\"/usr/lib/python3.7/site-packages/cherrypy/lib/encoding.py\\", line 221, in __call__\\n    self.body = self.oldhandler(*args, **kwargs)\\n  File \\"/usr/lib/python3.7/site-packages/cherrypy/_cptools.py\\", line 237, in wrap\\n    return self.newhandler(innerfunc, *args, **kwargs)\\n  File \\"/ceph/src/pybind/mgr/dashboard/services/exception.py\\", line 88, in dashboard_exception_handler\\n    return handler(*args, **kwargs)\\n  File \\"/usr/lib/python3.7/site-packages/cherrypy/_cpdispatch.py\\", line 60, in __call__\\n    return self.callable(*self.args, **self.kwargs)\\n  File \\"/ceph/src/pybind/mgr/dashboard/controllers/__init__.py\\", line 541, in inner\\n    ret = func(*args, **kwargs)\\n  File \\"/ceph/src/pybind/mgr/dashboard/controllers/__init__.py\\", line 734, in wrapper\\n    return func(*vpath, **params)\\n  File \\"/ceph/src/pybind/mgr/dashboard/controllers/mgr_modules.py\\", line 45, in get\\n    _backend = mgr.remote(\\"orchestrator_cli\\", \\"_select_orchestrator\\")\\n  File \\"/ceph/src/pybind/mgr/mgr_module.py\\", line 1187, in remote\\n    args, kwargs)\\nImportError: Module not found\\n", "version": "10.2.1"}']
```

Reply:
This is a general issue of the Mgr, it still does not know anything about disabled modules. That's why https://github.com/ceph/ceph/pull/25651 has been introduced to access module options for example.

Reply:
Use the following instead:
```
_backend = mgr.get_module_option_ex('orchestrator_cli', 'orchestrator')
```

Reply:
the `orchestrator_cli` module is always on. See the change in `src/mon/MgrMonitor.cc`

Reply:
OK, nevertheless i suggest to use the code above. In this case the ``_select_orchestrator`` method is obsolete.

Reply:
To be fair, the whole class `OrchClient` is a duplicate of `orchestrator.OrchestratorClientMixin` (with the exception of using `mgr` instead of `self`). But merging them is not ultra high priority for me. As `OrchClient` uses the same mechanism, I'd be :+1: for keeping them in sync

Reply:
> the `orchestrator_cli` module is always on. See the change in `src/mon/MgrMonitor.cc`

But this only means that this module is auto-started, but the admin can still disable it via CLI? In this case this is another reason to use my suggestion.

Reply:
> the `orchestrator_cli` module is always on. See the change in `src/mon/MgrMonitor.cc`

Sorry for hijacking this PR for something not related to this PR. How can we prevent that a user disables those always-on modules using #26116. Is there any info in the ``mgr_map`` that i can use for that? I don't want to keep them hardcoded in the Dashboard controller ``/api/mgr/module/<name>``.

Reply:
@votdev : I don't see where _select_orchestrator is being obsoleted. Can you elaborate? I can certainly change to use get_module_option_ex, but I'm unclear on why we'd want to do that instead of calling the orchestrator_cli method.

Also, do we want to disallow unplugging always-on modules? I'm not sure I see the value in doing that. IMO, if you decide to unplug it, then you get to keep both pieces when things break.

Reply:
> @votdev : I don't see where _select_orchestrator is being obsoleted. Can you elaborate? I can certainly change to use get_module_option_ex, but I'm unclear on why we'd want to do that instead of calling the orchestrator_cli method.
> 
T think you already know the answer because you've removed ``_select_orchestrator`` it in your last commit :-)

> Also, do we want to disallow unplugging always-on modules? I'm not sure I see the value in doing that. IMO, if you decide to unplug it, then you get to keep both pieces when things break.

That was just a thought of a potential problem. I don't know if it is possible to disable always-on modules, but i don't think the Mgr will prevent that. So this is a potential problem with the current implementation of this PR. If a admin accidentally disables the module, the ``mgr.remote`` call fails.